src/jbmc/jbmc_parse_options.cpp

```diff
@@ -625,14 +641,44 @@ int jbmc_parse_optionst::get_goto_program(
   return -1; // no error, continue
 }
 
-bool jbmc_parse_optionst::process_goto_program(
-  const optionst &options)
+void jbmc_parse_optionst::process_goto_function(
+  const irep_idt &function_name,
```

Comment:
`function_name` is unused

Reply:
It is in this implementation, but may be required by implementations that do other function-level passes that need access to function-related symbols.

Reply:
Are the actual uses thereof? Else it could easily be changed later on as the need arises, I'd say.

Reply:
'may be required' -- I don't think we should do something because it may be required. Anyways, this refactoring is outside the scope of this PR.

Reply:
There were passes that needed this. They are now not included in this PR.

Reply:
I'd be keen to see those as I'm not at all convinced they should be getting the function name in this way. I'd have recommended `get_function_id()`.

Reply:
If you're saying that we can get the function name from the `functiont` then this parameter isn't needed. Shall I remove it?

Reply:
You can get it from a non-empty `goto_programt`. If that's all you need (and once #1677 is merged) then you should be fine without that parameter.

Reply:
OK, removed.